screens/MapScreen/MapScreen.js

```diff
@@ -1,47 +1,157 @@
+import { YellowButton } from "../../components/Buttons/YellowButton";
+import { getDistance } from "geolib";
+import styles from "./MapScreenStyles";
+import Colors from "../../constants/colors";
+import monthNames from "../../constants/monthNames";
 
 export default class MapScreen extends React.Component {
+  static navigationOptions = () => {
+    return {
+      headerStyle: {
+        backgroundColor: Colors.green
+      }
+    };
+  };
+
+  constructor(props) {
+    super(props);
+    const region = this.props.navigation.getParam("region");
+    this.getUserPosition = this.getUserPosition.bind(this);
+    this.getTime = this.getTime.bind(this);
+    this.getOrdinalNum = this.getOrdinalNum.bind(this);
+    this.minutesWithLeadingZeros = this.minutesWithLeadingZeros.bind(this);
+    this.state = {
+      distance: 0,
+      userLatitude: 0,
+      userLongitude: 0,
+      giantLatitude: region.latitude,
+      giantLongitude: region.longitude,
+      giantName: this.props.navigation.getParam("name"),
+      giantDesc: this.props.navigation.getParam("desc"),
+      giantId: this.props.navigation.getParam("id"),
+      transport: this.props.navigation.getParam("transport"),
+      image: this.props.navigation.getParam("image"),
+      date: ""
+    };
+  }
+
+  componentDidMount() {
+    this.getUserPosition();
+    interval = setInterval(() => this.getUserPosition(), 3000);
+  }
+
+  getUserPosition() {
+    navigator.geolocation.getCurrentPosition(
+      position => {
+        const userLatitude = position.coords.latitude;
+        const userLongitude = position.coords.longitude;
+        this.setState({ userLatitude, userLongitude });
+      },
+      error => alert(error.message),
+      { enableHighAccuracy: true, timeout: 20000, maximumAge: 1000 }
+    );
+
+    let dis = getDistance(
+      {
+        latitude: this.state.giantLatitude,
+        longitude: this.state.giantLongitude
+      },
+      {
+        latitude: this.state.userLatitude,
+        longitude: this.state.userLongitude
+      }
+    );
+    this.setState({ distance: dis });
+
+    if (this.state.distance < 20000000 && this.state.distance !== 0) {
+      this.getTime();
+      this.props.navigation.navigate("RewardScreen", {
+        name: this.state.giantName,
+        desc: this.state.giantDesc,
+        giantId: this.state.giantId,
+        date: this.state.date,
+        image: this.state.image
+      });
+      clearInterval(interval);
+    }
+    this.setState({ isLoaded: true });
+  }
+
+  getTime() {
+    const now =
+      this.getOrdinalNum(new Date().getDate()) +
+      " OF " +
+      monthNames[new Date().getMonth()] +
+      " at " +
+      new Date().getHours() +
+      ":" +
+      this.minutesWithLeadingZeros(new Date().getMinutes());
+
+    this.setState({
+      date: now
+    });
+  }
+
+  getOrdinalNum(n) {
+    return (
+      n +
+      (n > 0
+        ? ["th", "st", "nd", "rd"][(n > 3 && n < 21) || n % 10 > 3 ? 0 : n % 10]
```

Comment:
Maybe you could turn some of these predicates into functions

Reply:
All logic is within this function so we will keep it as it is.